src/concrete/ml/quantization/post_training.py

```diff
@@ -23,6 +23,55 @@
 from .quantized_ops import QuantizedBrevitasQuant
 from .quantizers import QuantizationOptions, QuantizedArray, UniformQuantizer
 
+# pylint: disable=too-many-lines
+
+
+def get_n_bits_dict_trees(n_bits: Union[int, Dict[str, int]]) -> Dict[str, int]:
+    """Convert the n_bits parameter into a proper dictionary for tree based-models.
+
+    Args:
+        n_bits (int, Dict[str, int]): number of bits for quantization, can be a single value or
+            a dictionary with the following keys :
+            - "op_inputs" (mandatory)
+            - "op_leaves" (optional)
+            When using a single integer for n_bits, its value is assigned to "op_inputs" and
+            "op_leaves" bits.
+
+    Returns:
+        n_bits_dict (Dict[str, int]): A dictionary properly representing the number of bits to use
+            for quantization.
+    """
+
+    assert_true(
```

Comment:
by the way this lets me think that this should not be an `assert` but rather an ValueError raise (same goes for `get_n_bits_dict_trees`). I don't think asserting a user input is a great practice, but I guess we could make this change in a following PR

Reply:
I agree.

I'll do it in this PR for get_n_bits_trees, and do it for the other function in another PR

Reply:
yes great ! here's the issue for it : https://github.com/zama-ai/concrete-ml-internal/issues/4232